src/httpserver.cpp

```diff
@@ -459,15 +477,27 @@ void StopHTTPServer()
         evhttp_del_accept_socket(eventHTTP, socket);
     }
     boundSockets.clear();
-    if (eventBase) {
-        LogPrint(BCLog::HTTP, "Waiting for HTTP event thread to exit\n");
-        if (g_thread_http.joinable()) g_thread_http.join();
+    {
+        WAIT_LOCK(g_requests_mutex, lock);
+        g_requests_cv.wait(lock, []() EXCLUSIVE_LOCKS_REQUIRED(g_requests_mutex) {
+                if (!g_requests.empty()) {
+                    LogPrint(BCLog::HTTP, "Waiting for %d requests to stop HTTP server\n", g_requests.size());
```

Comment:
Having side effects in the predicate seems a bad idea due to spurious wakeups. It should be a pure function.

Reply:
Sure, then I think [this](https://github.com/bitcoin/bitcoin/pull/26742#discussion_r1057353934) isn't possible anymore so I am reverting back to how I did it before. I think that's fine though.